test/unit/clients/connector-client/connector-post-worldpay-check-3ds-flex-config.pact.test.js

```diff
@@ -0,0 +1,127 @@
+'use strict'
+
+const path = require('path')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+const { Pact } = require('@pact-foundation/pact')
+const expect = chai.expect
+chai.should()
+chai.use(chaiAsPromised)
+
+const PactInteractionBuilder = require('../../../fixtures/pact-interaction-builder').PactInteractionBuilder
+const worldpay3dsFlexCredentialsFixtures = require('../../../fixtures/worldpay-3ds-flex-credentials.fixtures')
+const Connector = require('../../../../app/services/clients/connector.client').ConnectorClient
+const PORT = Math.floor(Math.random() * 48127) + 1024
+const connectorClient = new Connector(`http://localhost:${PORT}`)
+const ACCOUNTS_RESOURCE = '/v1/api/accounts'
+const EXISTING_GATEWAY_ACCOUNT_ID = 1
+const CHECK_WORLDPAY_3DS_FLEX_CREDENTIALS = 'worldpay/check-3ds-flex-config'
+
+describe('connector client - check Worldpay 3DS Flex credentials', () => {
+  const provider = new Pact({
+    consumer: 'selfservice',
+    provider: 'connector',
+    port: PORT,
+    log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+    dir: path.resolve(process.cwd(), 'pacts'),
+    spec: 2,
+    pactfileWriteMode: 'merge'
+  })
+
+  before(() => provider.setup())
+  afterEach(() => provider.verify())
+  after(() => provider.finalize())
+
+  describe('when a request to check Worldpay 3DS Flex credentials is made', () => {
+    describe('and the credentials are valid and pass the existing format validation', () => {
+      const checkValidWorldpay3dsFlexCredentialsRequest = worldpay3dsFlexCredentialsFixtures.checkValidWorldpay3dsFlexCredentialsRequest()
+      const checkValidWorldpay3dsFlexCredentialsResponse = worldpay3dsFlexCredentialsFixtures.checkValidWorldpay3dsFlexCredentialsResponse()
+      before(done => {
+        const pactifiedRequest = checkValidWorldpay3dsFlexCredentialsRequest.getPactified()
+        const pactifiedResponse = checkValidWorldpay3dsFlexCredentialsResponse.getPactified()
+
+        provider.addInteraction(
+          new PactInteractionBuilder(`${ACCOUNTS_RESOURCE}/${EXISTING_GATEWAY_ACCOUNT_ID}/${CHECK_WORLDPAY_3DS_FLEX_CREDENTIALS}`)
+            .withState('has the credentials')
```

Comment:
This state should be more specific. I suggest `a gateway account with Worldpay 3DS Flex credentials exists`. Then, in pay-connector's [ContractTest](https://github.com/alphagov/pay-connector/blob/master/src/test/java/uk/gov/pay/connector/pact/ContractTest.java#L72-L73), there should be a method like 
```
    @State("a gateway account with Worldpay 3DS Flex credentials exists")
    public void aGatewayAccountWithWorldpay3dsFlexCredentialsExists(Map<String, String> params) {
        // add gatweay account with worldpay 3ds flex creds to the database
    }
```
Then, a pull request should be raised against pay-connector first before this selfservice PR can pass.

Reply:
Thanks @oswaldquek. I will make the changes.

Reply:
Done. Please see the [pull request](https://github.com/alphagov/pay-connector/pull/2754) on pay-connector.